### PR TITLE
Map operator

### PR DESCRIPTION
**Pull request by davidwdan**
fixes https://github.com/asm89/Rx.PHP/issues/10

### CODE REVIEW

**Comment by davidwdan** [comment]
I'm also going to change where to filter

**Comment by asm89** [comment]
@davidwdan Can you do where -> filter in a different PR?

File: lib/Rx/Operator/FilterOperator.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+
+namespace Rx\Operator;
+
+
```
Comment (asm89) [inline lib/Rx/Operator/FilterOperator.php]: Can you revert this in this PR? Also lots of unneeded whitespace here. :)

File: lib/Rx/Operator/MapOperator.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
+<?php
+
+
+namespace Rx\Operator;
+
+
```
Comment (asm89) [inline lib/Rx/Operator/MapOperator.php]: nit: can you get rid of this whitespace?

File: lib/Rx/Operator/MapOperator.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+use Rx\ObservableInterface;
+use Rx\Observer\CallbackObserver;
+use Rx\ObserverInterface;
+use Rx\SchedulerInterface;
+
+class MapOperator implements OperatorInterface
+{
+
```
Comment (asm89) [inline lib/Rx/Operator/MapOperator.php]: this as well

File: lib/Rx/Operator/MapOperator.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    /** @var callable  */
+    private $selector;
+
+    /**
+     * MapOperator constructor.
+     * @param $selector
+     */
+    public function __construct($selector)
+    {
+        if ( ! is_callable($selector)) {
```
Comment (asm89) [inline lib/Rx/Operator/MapOperator.php]: You can add a `callable` typehint to the constructor and remove this check as we're PHP 5.4+ now.

Comment (asm89) [inline lib/Rx/Operator/MapOperator.php]: Nvm. Let's keep it here and fix it in one sweep. Filed an issue for it: https://github.com/asm89/Rx.PHP/issues/14.

**Pull request merged**